### PR TITLE
feat: added GET `/currencies`, `/currencies/{isoCode}` and `/currencies/exchange` endpoints

### PR DESCRIPTION
**Pull request by IgnatG**
### Changelog
- added `/currencies` endpoint
- added `/currencies/{isoCode}` endpoint
- added `/currencies/exchange` endpoint
- added API and unit tests
- added Database names to constants

### CODE REVIEW

File: test/currencies/currencies.api-test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      const { status, body } = await api.get('/currencies/exchange?source=GBP');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?target=GBP`', async () => {
```
Comment (abhi-markan) [inline test/currencies/currencies.api-test.ts:103]: Can we please add a test case where both `source` and `target` and empty?

File: test/currencies/currencies.api-test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            effectiveTo: expect.any(String),
+            acbsCode: expect.any(String),
+          }),
+        ]),
+      );
+    });
+  });
+
+  describe('/currencies/{isoCode}', () => {
+    it('should return 200 on GET `/currencies/GBP`', async () => {
```
Comment (abhi-markan) [inline test/currencies/currencies.api-test.ts:136]: Test case where endpoint invoked is `/currencies/` please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, only few comments to address please.

File: package.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
   "license": "MIT",
   "author": "UKEF",
   "scripts": {
+    "api-test": "jest --selectProjects=API --detectOpenHandles --forceExit",
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply package.json:9]: we have a package called `sort-package-json` that standardizes the package.json file. this is done automatically.

File: src/modules/currencies/currencies.service.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+  async findOne(isoCode: string): Promise<CurrencyEntity[]> {
+    try {
+      const results = await this.currency.find({
+        where: { effectiveTo: Equal(DATE.MAXIMUM_LIMIT), isoCode },
+        order: { id: 'ASC' },
+      });
+
+      if (!results.length) {
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply src/modules/currencies/currencies.service.ts:38]: it can't be `undefined` - it's either an empty array or an array with values.

File: src/modules/currencies/currencies.service.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        throw new InternalServerErrorException();
+      }
+    }
+  }
+
+  async findExchangeRate(source: string, target: string, exchangeRateDate?: string): Promise<CurrencyExchangeEntity[]> {
+    try {
+      const results = await this.currencyExchange.query('USP_READ_CURRENCY_EXCHANGE_RATE @0, @1, @2', [source, target, exchangeRateDate]);
+
+      if (!results.length) {
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply src/modules/currencies/currencies.service.ts:58]: the result will always be an empty array or an array with values - so `undefined` won't be returned

File: test/currencies/currencies.api-test.ts (hunk 2 above)
Comment (IgnatG) [reply test/currencies/currencies.api-test.ts:136]: that's already covered above?

File: test/currencies/currencies.api-test.ts (hunk 1 above)
Comment (IgnatG) [reply test/currencies/currencies.api-test.ts:113]: I've added this above

File: src/constants/database-name.constant.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const DATABASE = {
+  CEDAR: 'mssql-cedar',
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply src/constants/database-name.constant.ts:2]: nope - not needed. we just gave them a label so we differentiate between them

File: src/modules/currencies/currencies.controller.test.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+import { CurrenciesController } from './currencies.controller';
+import { CurrenciesService } from './currencies.service';
+import { GetCurrencyExchangeDto } from './dto';
+
+describe('CurrenciesController', () => {
+  let currencyController: CurrenciesController;
+  let currencyService: CurrenciesService;
+
+  beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply src/modules/currencies/currencies.controller.test.ts]: updated

File: src/constants/index.ts [hunk 8]
```diff
@@ -3,6 +3,8 @@
  * Following constants are served:
  *
  * 1. Date
+ * 1. Database name
```
[the remark this answers is not in the document]
Comment (IgnatG) [reply src/constants/index.ts:6]: good spot - updated now

File: src/modules/currencies/currencies.controller.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  @ApiResponse({
+    status: 200,
+    description: 'Get the Active exchange rate',
+    type: GetCurrencyExchangeDto,
+  })
+  @ApiQuery({
+    name: 'source',
+    required: true,
+    type: 'string',
+    description: 'Source currency for exchange rate - Use ISO 3 alpha currency code standard',
```
Comment (avaitonis) [inline src/modules/currencies/currencies.controller.ts:37]: We support just 2 source currencies, GBP and USD. Maybe this should be part of swagger documentation and/or validation.

Comment (IgnatG) [reply src/modules/currencies/currencies.controller.ts]: good shout - updated now

**Review by avaitonis — APPROVED** [review]
LGTM

**Review by abhi-markan — APPROVED** [review]
